src/EmbedIO/Files/FileSystemProvider.cs

```diff
@@ -28,7 +32,7 @@ public class FileSystemProvider : IDisposable, IFileProvider
         public FileSystemProvider(string fileSystemPath, bool isImmutable)
         {
             FileSystemPath = Validate.LocalPath(nameof(fileSystemPath), fileSystemPath, true);
-            IsImmutable = isImmutable;
+            IsImmutable = isImmutable && !RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
```

Comment:
`IsImmutable = isImmutable || RuntimeInformation.IsOSPlatform(OSPlatform.OSX);`

If we cannot detect changes, we must treat the folder as immutable.

This introduces an usage restriction under OS/X that should be mentioned somewhere in the docs.

Reply:
I did it in the remarks of the constructor.

Reply:
I'm adding to the Extension method too.

Reply:
Done!